MultiplexReservationSystem/src/main/java/p/lodz/pl/multiplexreservationsystem/model/Tickets.java

```diff
@@ -0,0 +1,29 @@
+package p.lodz.pl.multiplexreservationsystem.model;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import lombok.*;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode
+public class Tickets {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "reservation_id")
+  @JsonBackReference
+  @EqualsAndHashCode.Exclude
+  private Reservations reservation;
+  @NotNull
+  private long seatId;
+  @NotNull
+  private TicketType type;
```

Comment:
To jest typ biletu, a klasa ma liczbę mnogą?

Reply:
Jak każdy dobry programista staram się zachować jednolitość w konwencji nazewniczej. Cały projekt zacząłem od opracowania diagramu ERD (jest w rootcie repozytorium, na branchu develop).  Następnie napisałem data.sql i schema.sql. Encje zazwyczaj powinniśmy nazywać rzeczownikami w liczbie mnogiej. Po przejściu do trzeciegu etapu, czyli do implementacji klas modelu dowiedziałem się, że nazwy w schema.sql muszą się pokrywać. W językach programistycznych zazwyczaj staramy się nazywać klasy inaczej, tj. rzeczownikiem l. poj. Następuje tu konflikt, który z początku nie raził aż tak w oczy. W obecnym stanie rzeczywiście nie wygląda to dobrze, ale niestety po krótkim refaktoringu nazw klas z nazwami tabel w schema.sql aplikacja nie działała poprawnie. Czułem na sobie presje czasu, bo termin oddania projektu dobiegał końca i niestety musiałem poświęcić ten problem na rzecz nauki Mock testów.